From b21e0b90ccb99a377bce0167fed1e881bb5065d7 Mon Sep 17 00:00:00 2001
From: Zhang Yanfei <zhangyanfei@cn.fujitsu.com>
Date: Fri, 22 Feb 2013 16:35:48 -0800
Subject: [PATCH] vmscan: change type of vm_total_pages to unsigned long
Git-commit: b21e0b90ccb99a377bce0167fed1e881bb5065d7
Patch-mainline: 3.9-rc1
References: bnc#864058

This variable is calculated from nr_free_pagecache_pages so
change its type to unsigned long.

Signed-off-by: Zhang Yanfei <zhangyanfei@cn.fujitsu.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Acked-by: Michal Hocko <mhocko@suse.cz>

---
 include/linux/swap.h |    2 +-
 mm/vmscan.c          |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/include/linux/swap.h
+++ b/include/linux/swap.h
@@ -264,7 +264,7 @@ extern void shrink_page_cache(gfp_t mask
 extern unsigned int vm_pagecache_limit_mb;
 extern unsigned int vm_pagecache_ignore_dirty;
 extern int remove_mapping(struct address_space *mapping, struct page *page);
-extern long vm_total_pages;
+extern unsigned long vm_total_pages;
 
 #ifdef CONFIG_NUMA
 extern int zone_reclaim_mode;
--- a/mm/vmscan.c
+++ b/mm/vmscan.c
@@ -130,7 +130,7 @@ struct scan_control {
 int vm_swappiness __read_mostly = 60;
 unsigned int vm_pagecache_limit_mb __read_mostly = 0;
 unsigned int vm_pagecache_ignore_dirty __read_mostly = 1;
-long vm_total_pages __read_mostly;	/* The total number of pages which the VM controls */
+unsigned long vm_total_pages __read_mostly;	/* The total number of pages which the VM controls */
 
 static LIST_HEAD(shrinker_list);
 static DECLARE_RWSEM(shrinker_rwsem);
